app/controllers/organizations/staff/invitations_controller.rb

```diff
@@ -1,111 +1,126 @@
-      )
-      @user.add_role(user_params[:roles], Current.organization)
-
-      if @user.save
-        @user.invite!(current_user)
-        create_person(@user)
-        redirect_to staff_staff_index_path, notice: t(".success")
-      else
-        render "organizations/staff/staff_invitations/new", status: :unprocessable_entity
-      end
-    when "fosterer"
-      authorize! User, context: {organization: Current.organization},
-        with: Organizations::FostererInvitationPolicy
-
-      @user = User.new(
-        user_params.merge(password: SecureRandom.hex(8)).except(:roles)
-      )
-      @user.add_role("adopter", Current.organization)
-      @user.add_role("fosterer", Current.organization)
-
-      if @user.save
-        @user.invite!(current_user)
-        create_person(@user)
-        redirect_to staff_fosterers_path, notice: t(".success")
-      else
-        render "organizations/staff/fosterer_invitations/new", status: :unprocessable_entity
+      private
+
+      def user_params
+        params.require(:user)
+          .permit(
+            :first_name, :last_name, :email, :roles
+          )
       end
-    else
-      authorize! User, context: {organization: Current.organization},
-        with: Organizations::InvitationPolicy
 
-      redirect_back fallback_location: root_path
-    end
-  end
+      def configure_permitted_parameters
+        devise_parameter_sanitizer.permit(:accept_invitation, keys: %i[tos_agreement])
+      end
 
-  private
-
-  def user_params
-    params.require(:user)
-      .permit(
-        :first_name, :last_name, :email, :roles,
-        person_attributes: [
-          :phone_number,
-          location_attributes: %i[country province_state city_town]
-        ]
-      )
-  end
+      def after_accept_path_for(_resource)
+        # Devise sets `current_user` only after this callback runs, so we
+        # must explicitly call it here
+        set_current_person
 
-  def configure_permitted_parameters
-    devise_parameter_sanitizer.permit(:accept_invitation, keys: %i[tos_agreement])
-  end
+        if allowed_to?(
+          :index?, with: Organizations::DashboardPolicy
+        )
+          staff_dashboard_index_path
+        elsif allowed_to?(
+          :index?, with: Organizations::AdopterFosterDashboardPolicy
+        )
+          adopter_fosterer_dashboard_index_path
+        else
+          root_path
+        end
+      end
 
-  def after_accept_path_for(_resource)
-    if allowed_to?(
-      :index?, with: Organizations::DashboardPolicy,
-      context: {organization: Current.organization}
-    )
-      staff_dashboard_index_path
-    elsif allowed_to?(
-      :index?, with: Organizations::AdopterFosterDashboardPolicy,
-      context: {organization: Current.organization}
-    )
-      adopter_fosterer_dashboard_index_path
-    else
-      root_path
-    end
-  end
+      def create_person(user)
+        # TODO: do we need this condition?
```

Comment:
we could probably get rid of this if we used `find_or_create_by` in the `set_person` method in place of this method call `create_person`?